lib/DataFileUtil/DataFileUtilImpl.py

```diff
@@ -931,6 +934,67 @@ def unpack_file(self, ctx, params):
 
+    def unpack_files(self, ctx, params):
+        """
+        Using the same logic as unpacking a Shock file, this method will cause
+        any bzip or gzip files to be uncompressed, and then unpack tar and zip
+        archive files (uncompressing gzipped or bzipped archive files if 
+        necessary). If the file is an archive, it will be unbundled into the 
+        directory containing the original output file.
+        The ordering of the input and output files is preserved in the input and output lists.
+        :param params: instance of list of type "UnpackFilesParams" (Input
+           parameters for the unpack_files function. Required parameter:
+           file_path - the path to the file to unpack. The file will be
+           unpacked into the file's parent directory. Optional parameter:
+           unpack - either 'uncompress' or 'unpack'. 'uncompress' will cause
+           any bzip or gzip files to be uncompressed. 'unpack' will behave
+           the same way, but it will also unpack tar and zip archive files
+           (uncompressing gzipped or bzipped archive files if necessary). If
+           'uncompress' is specified and an archive file is encountered, an
+           error will be thrown. If the file is an archive, it will be
+           unbundled into the directory containing the original output file.
+           Defaults to 'unpack'. Note that if the file name (either as
+           provided by the user or by Shock) without the a decompression
+           extension (e.g. .gz, .zip or .tgz -> .tar) points to an existing
+           file and unpack is specified, that file will be overwritten by the
+           decompressed Shock file.) -> structure: parameter "file_path" of
+           String, parameter "unpack" of String
+        :returns: instance of list of type "UnpackFilesResult" (Output
+           parameters for the unpack_files function. file_path - the path to
+           the unpacked file, or in the case of archive files, the path to
+           the original archive file.) -> structure: parameter "file_path" of
+           String
+        """
+        # ctx is the context object
+        # return variables are: out
+        #BEGIN unpack_files
+        del ctx
+        # could do all the error checking first to avoid partial unpacks, but that doesn't
+        # seem like a major problem, so just do it Q&D for now.
+        out = []
+        if not params:
+            raise ValueError("Must provide at least one input parameter dictionary in a list")
+        for i, param in enumerate(params, start=1):
+            file_path = param.get('file_path')
+            if not file_path:
+                raise ValueError(f'Must provide file path for param #{i}')
+            unpack = param.get(_UNPACK)
+            if unpack not in [None, _UNPACK, _UNCOMPRESS]:
+                raise ValueError(f'Illegal unpack value for param #{i}: {unpack}')
+            unpack = unpack is None or unpack == _UNPACK
```

Comment:
we might have a situation where `unpack_files` fails but the first few files might actually unpack successfully. Is that an expected behavior?

Reply:
yeah, I noted that here: https://github.com/kbaseapps/DataFileUtil/pull/91/files#diff-ad387a693a38e713e0a0154089a38eb5a1334d60ebafb3597c44a0070a36b25aR972-R973

Even if there was an error checking loop prior to the unpack calls, it'd still be possible for partial unpacks to occur if a file was missing (unless we check for that too, duplicating work in the single unpack function) or was the incorrect format. 

I can't think of a reason that a partial unpack would be a problem - presumably the app run would fail and the scratch space would get cleaned up. Can you think of a reason we should be more careful here?

Reply:
yea. 👍  It probably won't cause any problems.